chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -539,18 +539,40 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
     val nameMap = LinkedHashMap[String, Data]()
     val seen = HashSet[Data]()
     for (m <- getPublicFields(classOf[Bundle])) {
-      getBundleField(m) foreach { d =>
-        if (nameMap contains m.getName) {
-          require(nameMap(m.getName) eq d)
-        } else if (!seen(d)) {
-          nameMap(m.getName) = d
-          seen += d
-        }
+      getBundleField(m) match {
+        case Some(d: Data) =>
+          if (nameMap contains m.getName) {
+            require(nameMap(m.getName) eq d)
+          } else if (!seen(d)) {
+            nameMap(m.getName) = d
+            seen += d
+          }
+        case None =>
+          if (!ignoreSeq) {
```

Comment:
I understand why this check is here, but is there a cleaner way to implement the bypass than setting a global flag (which seems a bit unclean).

Reply:
I'm happy to take suggestions, if you have any. This is actually a method that a user can override in their bundle and is read-only, so I don't think this can cause any concurrency issues?

Reply:
It’s not a global flag right? It’s a per-Bundle flag which is fairly contained.

Reply:
Sorry, global was bad terminology, I meant throwing flags in the body of the instance as override vals. I don't think this is something we've done elsewhere, and while it's mitigated by clear time-of-use documentation (which is great!), it still feels odd. I don't have any solutions right now, but it would be worth brainstorming alternatives at a future meeting.

Reply:
Yeah I agree that overriding a function isn't ideal. What if we made it a trait you could mixin instead?

Reply:
Done.